### ISSUE TITLE
Thought about submitting this to oh-my-fish to increase visibility?

### ISSUE THREAD

**Issue by iggy** [evanlucas/fish-kubectl-completions#11]
I'm not sure if I'm the weird one here, but I almost always look for fish things in `omf` before resorting to google. I think this package being available via omf would be awesome.

**Comment by xordspar0** [comment]
Or you could submit it upstream to the fish project itself.

**Comment by dotCipher** [comment]
Why would it be better to submit to the main fish project itself? If anything, it should probably be implemented in `kubectl` itself, not the core fish shell.

However it seems like both the issues raised in the `kubernetes` repo: https://github.com/kubernetes/kubernetes/issues/30763 as well as the `kubectl` repo: https://github.com/kubernetes/kubectl/issues/131 were closed without any reasonable solution.

The workaround here would probably be just to include it in fish package managers instead (like omf).

**Comment by xordspar0** [comment]
I does make more sense for kubectl to support completion for fish. There is progress in that area. https://github.com/spf13/cobra/pull/754 is an open PR to cobra, kubectl's CLI library, that would add this support. https://github.com/kubernetes/kubectl/issues/131 was closed due to inactivity, but the change to cobra would allow it to move forward.

**Comment by dotCipher** [comment]
Ah ok, I missed that one.  I stand corrected, thanks!

**Comment by jorgebucaran** [comment]
Hi @dotCipher, I don't know if this is unusual in other platforms, but in fish, the practice is to do as suggested and [submit completions to the main repository](https://github.com/fish-shell/fish-shell/pulls?utf8=%E2%9C%93&q=add+completions). The ideal would be if `kubectl` sent completions to fish-shell/fish-shell themselves, but there's nothing strange about users doing it.

**Comment by dotCipher** [comment]
Yeah, I misunderstood, but that makes sense now.  I guess we will have to wait on spf13/cobra#754 to finish before taking further action here.

**Comment by arbourd** [comment]
>I guess we will have to wait on spf13/cobra#754 to finish before taking further action here.

Why? Re: submit completions to the main repository

**Comment by wwentland** [comment]
> Why? Re: submit completions to the main repository

The underlying reason is that there is a preference for completions to be generated programmatically in a similar way to what is done for bash and zsh.
